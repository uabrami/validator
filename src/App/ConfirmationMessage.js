import React from "react";
import "./styles.css";

const ConfirmationMessage = ({ isValid, message }) =>{
  const isValidColor = isValid ? '#61B329' : '#DC143C';
 return (
    <p
      className='confirmationMessage'
      style={{color: isValidColor}}
    >
      {message}
    </p>
);
 }

export default ConfirmationMessage;