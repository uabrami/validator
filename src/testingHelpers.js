import { render } from "@testing-library/react";

export const checkBasicRendering = (component) => {
  it("Expect to render", () => {
    const { container } = render(component);
    expect(container.firstChild).toBeDefined();
  });
};

export const checkRequiredProps = (component) => {
  it("Expect to have all required propTypes", () => {
    const spy = jest.spyOn(global.console, "error");
    render(component);
    expect(spy).not.toHaveBeenCalled();
    spy.mockClear();
  });
};
