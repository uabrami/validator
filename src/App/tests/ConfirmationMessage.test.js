import React from "react";
import { render } from "@testing-library/react";

import ConfirmationMessage from "../ConfirmationMessage";

import { checkBasicRendering, checkRequiredProps } from "../../testingHelpers";

const testProps = {
  isValid: true,
  message: "test",
};

const Component = <ConfirmationMessage {...testProps} />;

describe("<ConfirmationMessage />", () => {
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });

  checkBasicRendering(Component);
  checkRequiredProps(Component);
});
