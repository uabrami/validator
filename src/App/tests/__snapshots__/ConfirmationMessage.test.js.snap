// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationMessage /> Should render and match the snapshot 1`] = `
<p
  class="confirmationMessage"
  style="color: rgb(97, 179, 41);"
>
  test
</p>
`;
